### PR TITLE
nrf_modem: doc: move socket topics to subfolder + cleanup

### PR DESCRIPTION
**Pull request by MirkoCovizzi**
This commit moves `PDN`, `TLS/DTLS`, and `RAI` from
`sockets.rst` to the `sockets` subfolder in their
own respective document page.

Also update `PDN` and `TLS/DTLS` documents.

### CODE REVIEW

**Comment by MirkoCovizzi** [comment]
@divipillai In this comment https://github.com/lemrey/sdk-nrfxlib/pull/33#discussion_r1453290869 you propose some examples of subpages. We discussed offline about the layout and we agreed on embedding the subpages instead.

File: nrf_modem/doc/sockets.rst [hunk 1]
```diff
@@ -10,326 +10,14 @@ Networking sockets
...
-
-.. include:: sockets/pdn.rst
-
-.. include:: sockets/tls_dtls.rst
-
-.. include:: sockets/rai.rst
+   sockets/socket_options_func
+   sockets/pdn
+   sockets/tls_dtls
+   sockets/rai
```
Comment (anhmolt) [inline nrf_modem/doc/sockets.rst:23]: I think I prefer this solution with sub-paging better than including/embedding the other rst files into sockets.rst. The "networking sockets" page looks clean and it is easy to see what topics the subpages address.

File: nrf_modem/doc/sockets/socket_options_func.rst [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+| NRF_IPPROTO_IP  | NRF_SO_IP_ECHO_REPLY            | ``int``                | get/set    | Non-zero enables ICMP echo replies on IPv4.                                                |
++-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
+| NRF_IPPROTO_IPV6| NRF_SO_IPV6_ECHO_REPLY          | ``int``                | get/set    | Non-zero enables ICMP echo replies on IPv6.                                                |
++-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
+| NRF_IPPROTO_TCP | NRF_SO_TCP_SRV_SESSTIMEO        | ``int``                | get/set    | Non-zero enables TCP server session timeout after a configurable period of inactivity.     |
++-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
+
+The details for each socket are described as follows:
```
Comment (anhmolt) [inline nrf_modem/doc/sockets/socket_options_func.rst:101]: Is this sentence correctly talking of sockets or should it be socket options?
```Suggestion
The details for each socket option are described as follows:
```

File: nrf_modem/doc/sockets/socket_options_func.rst [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
++-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
+| NRF_IPPROTO_IP  | NRF_SO_IP_ECHO_REPLY            | ``int``                | get/set    | Non-zero enables ICMP echo replies on IPv4.                                                |
++-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
+| NRF_IPPROTO_IPV6| NRF_SO_IPV6_ECHO_REPLY          | ``int``                | get/set    | Non-zero enables ICMP echo replies on IPv6.                                                |
++-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
+| NRF_IPPROTO_TCP | NRF_SO_TCP_SRV_SESSTIMEO        | ``int``                | get/set    | Non-zero enables TCP server session timeout after a configurable period of inactivity.     |
```
Comment (anhmolt) [inline nrf_modem/doc/sockets/socket_options_func.rst:98]: These four options are placed above the `NRF_SO_SEC_xxx` options in the details section below the table, but they are at the end of this table. Not super important, but would it be good to align the order they appear in this table and in the list below?

**Review by anhmolt — APPROVED** [review]

File: nrf_modem/doc/sockets/socket_options_func.rst (hunk 2 above)
Comment (MirkoCovizzi) [reply nrf_modem/doc/sockets/socket_options_func.rst:101]: I can make an extra commit for the fixes on top of this refactor.

File: nrf_modem/doc/sockets/socket_options_func.rst (hunk 3 above)
Comment (MirkoCovizzi) [reply nrf_modem/doc/sockets/socket_options_func.rst:98]: Might be better to make a dedicated task for this as it strays from the subject of this PR.

**Review by divipillai — APPROVED** [review]

**Comment by MirkoCovizzi** [comment]
Discussed offline with the team regarding the choice of splitting the layout into subpages. The benefit is that users that want to read RAI, for example, now don't have to scroll all the way down to reach the topic. The users now can reach the topic by clicking the respective link in `sockets.rst`.

File: nrf_modem/doc/sockets/pdn.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+During operation, an active PDN connection may be deactivated due to loss of connectivity or other reasons.
+When a socket operation is attempted on a socket that no longer has an active PDN connection, the operation will return ``-1`` and set ``errno`` to ``NRF_ENETDOWN``.
+If the socket is being polled, the :c:func:`nrf_poll` function will set the ``POLLERR`` flag and set the socket error to ``NRF_ENETDOWN``.
+The socket error can be retrieved using the :c:macro:`NRF_SO_ERROR` socket option.
+
+When the ``NRF_ENETDOWN`` error is detected, the socket is no longer usable and must be closed by the application.
+The application is responsible for detecting when the PDN connection is activated again, before re-creating the socket and attempting the failed operation again.
+
+The packet domain AT commands can be used to manage packet data networks.
+Alternatively, the :ref:`pdn_readme` library in |NCS| can be used to receive events on PDN connectivity and manage packet data networks.
```
[the remark this answers is not in the document]
Comment (lemrey) [reply nrf_modem/doc/sockets/pdn.rst:61]: I agree wit this

**Review by lemrey — COMMENTED** [review]
One small comment

File: nrf_modem/doc/sockets/pdn.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (MirkoCovizzi) [reply nrf_modem/doc/sockets/pdn.rst:61]: I can make an extra commit for this.

File: nrf_modem/doc/sockets/pdn.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+   :depth: 1
+
+The Modem library supports selecting which Packet Data Network (PDN) to use on a network socket and for DNS queries.
+The configuration of Packet Data Protocol (PDP) contexts, and the activation of PDN connections are not handled by the Modem library.
+To configure PDP contexts and activate PDN connections, the application must use the packet domain AT commands.
+
+When performing network operations on any PDN, the application ensures that the PDN connection is available.
+For more information about how to configure PDP contexts, activate PDN connections, and determine their state, see the `Packet domain commands`_ section in the nRF9160 AT Commands Reference Guide or the same section in the `nRF91x1 AT Commands Reference Guide`_ depending on the SiP you are using.
+
+The packet domain AT commands can be used to manage packet data networks.
```
Comment (lemrey) [inline nrf_modem/doc/sockets/pdn.rst:17]: I think this sentence is redundant with the one above, we can remove it.

Comment (MirkoCovizzi) [reply nrf_modem/doc/sockets/pdn.rst]: Removed.

**Review by eivindj-nordic — APPROVED** [review]

**Pull request merged**